changelog.d/20241101_155044_chris_function_description_from_docstring.rst

```diff
@@ -0,0 +1,5 @@
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- Function docstrings are now read and used as the description for the function when it
+  is uploaded. This will support future UI changes to the webapp.
```

Comment:
This is a cool enough detail that I think we should document this beyond the changelog.  The first places that come to mind are:

- `sdk.rst`&nbsp;&mdash;&nbsp;a number of spots need to be updated.  Certainly any function we define should "show good habits and proper procedure," if you will.  And a couple of call sites need to be updated.
- We don't present the `WebClient` in our documentation, but I think we could stand to update the shim overtop `.register_function()` in the `Client`.

There may be other documentation details the present themselves when going through it, but these are just the first that come readily to sight and mind.

Reply:
On the other hand don't we return the description in the [GET function route](https://compute.api.globus.org/redoc#tag/Functions/operation/get_function_v2_functions__function_uuid__get), as implemented via `.get_function(...)`?

(Certainly not a nice UI -- can one even call it that?! --  but we _do_ share it in at least one place, currently.)